packages/flutter/lib/src/widgets/form.dart

```diff
@@ -430,8 +436,8 @@ class FormFieldState<T> extends State<FormField<T>> {
 
   @override
   Widget build(BuildContext context) {
-    // Only autovalidate if the widget is also enabled
-    if (widget.autovalidate && widget.enabled)
+    // Only autovalidate if the widget is also enabled and state value has changed at least once
+    if (widget.autovalidate && widget.enabled && _isStateDirty)
```

Comment:
There is also a setValue method, it will be called when the the form is built with new text editing controller. In that case you might want to reset this state dirty to false

Reply:
done the changes

Reply:
yes, this PR tries to achieve the same thing as the other one you are referring to. It's just that this PR was raised pretty earlier than the other one.